### PR TITLE
feat: support for ocp v4.15

### PR DESCRIPTION
**Pull request by Aashiq-J**
### Release required?

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [x] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

##### Release notes content

New feature : https://community.ibm.com/community/user/cloud/blogs/cale-rath/2024/03/07/secure-by-default-cluster-networking

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -241,6 +242,12 @@ variable "access_tags" {
   }
 }
 
+variable "disable_outbound_traffic_protection" {
+  type        = bool
+  description = "Whether to allow public outbound access from the cluster workers."
```
Comment (ocofaigh) [inline variables.tf:247]: you should mention this is only applicable for 4.15

**Review by ocofaigh — CHANGES_REQUESTED** [review]
please update the tests accordingly so they test 4.15

**Comment by vburckhardt** [comment]
Couple of invalid version error
```
TestRunUpgradeAdvancedExample 2024-05-03T08:35:17Z retry.go:99: Returning due to fatal error: FatalError{Underlying: error while running command: exit status 1; ╷
│ Error: Invalid value for variable
│ 
│   on main.tf line 160, in module "ocp_base":
│  160:   ocp_version          = var.ocp_version
│     ├────────────────
│     │ var.ocp_version is "4.15"
│ 
│ The specified ocp_version is not of the valid versions.
│ 
│ This was checked by the validation rule at ../../variables.tf:156,3-13
```.

**Comment by Aashiq-J** [comment]
@vburckhardt , @ocofaigh 
Should i skip the upgrade test for this PR, the error is due to the pipeline running example with version 4.15 in main branch which doesn't have the updated validation.

**Review by vburckhardt — APPROVED** [review]

**Pull request merged**